src/app/component/comments/components/delete-comment/delete-comment.component.ts

```diff
@@ -9,16 +11,35 @@ import { CommentsDTO } from '../../models/comments-model';
 })
 export class DeleteCommentComponent {
   @Input() public element: CommentsDTO;
+  @Input() public dataType: string;
   @Output() public elementsList = new EventEmitter();
   public deleteIcon = 'assets/img/comments/delete.png';
 
-  constructor(private commentsService: CommentsService) { }
+  constructor(private commentsService: CommentsService,
+              private dialog: MatDialog) { }
 
-  public deleteComment(): void {
-    this.commentsService.deleteComments(this.element.id).subscribe(response => {
-      if (response.status === 200) {
-        this.elementsList.emit();
+  public openPopup(): void {
+    const dialogRef = this.dialog.open(WarningPopUpComponent, {
+      hasBackdrop: true,
+      closeOnNavigation: true,
+      disableClose: true,
+      panelClass: 'popup-dialog-container',
+      data: {
+        popupTitle: `homepage.eco-news.comment.${this.dataType}-popup.title`,
+        popupConfirm: `homepage.eco-news.comment.${this.dataType}-popup.confirm`,
+        popupCancel: `homepage.eco-news.comment.${this.dataType}-popup.cancel`,
       }
     });
+
+    const subscribeDialog = dialogRef.afterClosed().subscribe(confirm => {
+      if (confirm) {
+        this.commentsService.deleteComments(this.element.id).subscribe(response => {
+          if (response.status === 200) {
+            this.elementsList.emit();
+          }
+        });
+      }
+      subscribeDialog.unsubscribe();
```

Comment:
if i understand this correctly, is subscribeDialog designed to initialize a component? 
you subscribe, get one value and then you immediately unsubscribe ? 
if yes then consider using : take(1) or takeUntil() operators